### PR TITLE
Enable doctest and docstyle

### PR DESCRIPTION
**Pull request by beeender**
- Enable doctest in tox's test target, the examples in the source code
  will be a part of the test. Only examples in 'group.py' are tested for
  now. Others should be enabled by separated PRs.
- Enable docstyle check on GitHub actions, which will be executed
  before generating doc. Only 'group.py' and 'dataframe.py' are checked
  for now. Others should be enabled later.
- Rename 'DataFrameGroupingSets' to 'DataFrameGroupingSet'.
- Doc updates for 'group.py'.
- 'DataFrameGroupingSet.flattern()' returns a list since we need to keep
  a stable columns order for the test cases.

### CODE REVIEW

File: greenplumpython/group.py [hunk 1]
```diff
@@ -74,15 +113,17 @@ def assign(self, **new_columns: Callable[["DataFrame"], Any]) -> "DataFrame":
                 >>> rows = [(1,) for _ in range(10)]
                 >>> numbers = db.create_dataframe(rows=rows, column_names=["val"])
                 >>> count = gp.aggregate_function("count")
-                >>> results = numbers.group_by().assign(count=lambda t: count(t["val"]))
+                >>> sum = gp.aggregate_function("sum")
+                >>> results = numbers.group_by("is_even").assign(
```
Comment (Quasimewdo) [inline greenplumpython/group.py:117]: `numbers` doesn't have column `"is_even"`, the output seems not correct

File: greenplumpython/group.py [hunk 2]
```diff
@@ -101,20 +142,51 @@ def union(
...
+                >>>
+                >>> results = (
+                ...     numbers.group_by("is_even")
+                ...     .union(lambda t: t.group_by("is_multiple_of_3"))
+                ...     .assign(count=lambda t: count(t["val"]))
+                ... )
+                >>>
+                >>> results
+                ------------------------------------
+                 is_even | is_multiple_of_3 | count
```
[the remark this answers is not in the document]
Comment (beeender) [reply greenplumpython/group.py:174]: doctest looks nice, checking it ...

[the remark this answers is not in the document]
Comment (beeender) [reply greenplumpython/group.py:176]: Add doctest to make sure the examples pass. But the similar tests in test are kept for now. I think the cases in example and in tests are created for different purposes, duplicating some of them are acceptable for now.

**Comment by beeender** [comment]
@yihong0618 NICE and THANKS!

File: greenplumpython/group.py [hunk 3]
```diff
@@ -98,35 +151,66 @@ def assign(self, **new_columns: Callable[["DataFrame"], Any]) -> "DataFrame":
...
+                >>> numbers = db.create_dataframe(rows=rows, column_names=["val", "is_even", "is_multiple_of_3"])
+                >>> count = gp.aggregate_function("count")
+                >>> results = (
+                ...     numbers.group_by("is_even")
+                ...     .union(lambda t: t.group_by("is_multiple_of_3"))
+                ...     .assign(count=lambda t: count(t["val"]))
+                ... )
+                >>> results.order_by("is_even")[:].order_by("count")[:]
+                ------------------------------------
+                 is_even | is_multiple_of_3 | count
```
Comment (yihong0618) [inline greenplumpython/group.py:184]: seems we can support order_by muti-columns like .order_by(["a", "b"])?

**Review by yihong0618 — APPROVED** [review]

File: greenplumpython/group.py (hunk 3 above)
Comment (yihong0618) [reply greenplumpython/group.py:184]: @xuebinsu @beeender

**Review by Quasimewdo — APPROVED** [review]
LGTM

File: greenplumpython/group.py [hunk 4]
```diff
@@ -98,35 +151,66 @@ def assign(self, **new_columns: Callable[["DataFrame"], Any]) -> "DataFrame":
...
-        item_set: MutableSet[Expr] = set()
+        item_list: List[str] = list()
+        # Remove the duplicates and keep the input order
         for grouping_set in self._grouping_sets:
             for item in grouping_set:
                 assert isinstance(item, str), f"Grouping item {item} is not a column name."
-                item_set.add(item)
-        return item_set
+                if item not in item_list:
+                    item_list.append(item)
```
[the remark this answers is not in the document]
Comment (beeender) [reply greenplumpython/group.py]:
```
    def execute(self, query, vars=None):
        self.column_mapping = []
        self._query_executed = True
>       return super().execute(query, vars)
E       psycopg2.errors.SyntaxError: zero-length delimited identifier at or near """"
E       LINE 1: ..."."val")  ,"cte_8c07c676230a4def8e7e907bcbe6e53b".""is_even"...
```

This is not correct I think. Can you add quotes around column names?

[the remark this answers is not in the document]
Comment (Quasimewdo) [reply greenplumpython/group.py:207]: I think we can add quotes around column names.  I see two quotes here, is it correct? `.""is_even"`

[the remark this answers is not in the document]
Comment (beeender) [reply greenplumpython/group.py:207]: no, it is not, Any other places possibly adds another quote?

**Pull request merged**